libsolidity/formal/CHC.cpp

```diff
@@ -1001,6 +1007,37 @@ set<unsigned> CHC::transactionVerificationTargetsIds(ASTNode const* _txRoot)
 	return verificationTargetsIds;
 }
 
+optional<CHC::CHCNatspecOption> CHC::natspecOptionFromString(string const& _option)
+{
+	static map<string, CHCNatspecOption> options{
+		{"abstract-function-nondet", CHCNatspecOption::AbstractFunctionNondet}
+	};
+	if (options.count(_option))
+		return options.at(_option);
+	return {};
+}
+
+set<CHC::CHCNatspecOption> CHC::smtNatspecTags(FunctionDefinition const& _function)
+{
+	set<CHC::CHCNatspecOption> options;
+	auto const& tags = _function.annotation().docTags;
+	string smtStr = "custom:smtchecker";
+	for (auto it = tags.find(smtStr); it != tags.end() && it->first == smtStr; ++it)
```

Comment:
Can you reuse `NatSpec::extractCustomDoc` here?

Reply:
I'll check, thanks!

Reply:
I checked that function, and I don't think we gain anything by using it. I think the loop that's there is still going to be required.

Reply:
Exactly. It returns a multimap, so I need the loop anyway to parse all?

Reply:
Also: yea, I can see how we might in the future have multiple `@custom:smtchecker` tags with different values, so I'd treat them separately